source/extensions/filters/network/ratelimit/config.cc

```diff
@@ -42,8 +42,8 @@ Network::FilterFactoryCb RateLimitConfigFactory::createFilterFactoryFromProtoTyp
 /**
  * Static registration for the rate limit filter. @see RegisterFactory.
  */
-REGISTER_FACTORY(RateLimitConfigFactory,
-                 Server::Configuration::NamedNetworkFilterConfigFactory){"envoy.ratelimit"};
+REGISTER_FACTORY(RateLimitConfigFactory, Server::Configuration::NamedNetworkFilterConfigFactory){
+    "envoy.filters.network.rate_limit"};
```

Comment:
I don't think we can out-right change the name atomically, as I believe that would break existing configurations. Someone can correct me if I'm wrong, but I think both names need to be registered to allow a transition period.

Reply:
Now that we load based on typed config (and not name) I think this might be OK, but I'm not 100% sure. If it's going to break users honestly I'm not sure it's worth fixing this.

Reply:
It's very likely to break users who use type URL to identify an extension. I think the right procedure here is to register under dual names, and mark one as deprecated for some time.